### PR TITLE
Add a REST Controller and expose the page builder at it's own route

### PR DESCRIPTION
**Pull request by joehoyle**
This means one can get the page builder elements in it's own route for
isloation / testing and visibility.

### CODE REVIEW

File: inc/class-rest-controller.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_Error;
+use WP_Http;
+use WP_REST_Posts_Controller;
+
+/**
+ * REST API Controller for the page builder routes.
+ */
+class REST_Controller extends \WP_REST_Controller {
```
Comment (roborourke) [inline inc/class-rest-controller.php:15]: May as well `use WP_REST_Controller;` too but obvs no biggy

File: inc/class-rest-controller.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	}
+
+	public function get_items( $request ) {
+		$modules = $this->builder->get_raw_data( $request['id'] );
+
+		$num = 0;
+
+		$data = array_map( function ( $module ) use ( $request, &$num ) {
+			$module = Plugin::get_instance()->init_module( $module['name'], $module );
+			$response = $this->prepare_item_for_response( $module, $request );
```
Comment (roborourke) [inline inc/class-rest-controller.php:57]: Using `$this` here pins the plugin to PHP 5.4+, ok in my book but we should maybe make a note in the readme or plugin header about it.

File: inc/class-rest-controller.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+			$data = $response->get_data();
+			$data['id'] = $num;
+			$num++;
```
Comment (roborourke) [inline inc/class-rest-controller.php:60]: Could do the `++` on the line above as it'll happen after the value is read though I guess the intent is a bit less obvious then.

**Comment by roborourke** [comment]
No blockers here but a couple of notes worth considering.

File: inc/class-rest-controller.php (hunk 1 above)
Comment (joehoyle) [inline inc/class-rest-controller.php:15]: Oh yeah oops!

File: inc/class-rest-controller.php (hunk 2 above)
Comment (joehoyle) [inline inc/class-rest-controller.php:57]: Oh good point, hmm we are using namespaces so php 5.3 right now if we aren't something like this elsewhere. I'd be ok with requiring 5.4 (this is a modern plugin) I'll add a note to the readme.

File: inc/class-rest-controller.php (hunk 3 above)
Comment (joehoyle) [inline inc/class-rest-controller.php:60]: Yeah I think I prefer it like this.

**Pull request closed without merging**